### ISSUE TITLE
Documentation needed for `model_config`

### ISSUE THREAD

**Issue by cluhmann** [pymc-labs/pymc-marketing#1179]
For both MMM and CLV, the model constructors take a config dict called `model_config`. I can find examples this dict in various notebooks, but I cannot find any specification about what keys/values are valid/invalid.  Searching for "model_config" in the docs returns the `model_config` class, which contains nothing but [parse_model_config](https://www.pymc-marketing.io/en/stable/api/generated/pymc_marketing.model_config.parse_model_config.html#pymc_marketing.model_config.parse_model_config) which provides more examples (and a rather unhelpful invalid example) , but no specification.  Is this documented anywhere? If so, it should be pointed to in the documentation more thoroughly. If not, it should be documented.

I also see that there are other (potentially nested) config dicts like `intercept_tvp_config`. Are the specifications of those documented anywhere?

**Comment by wd60622** [comment]
How about here: https://www.pymc-marketing.io/en/stable/notebooks/general/model_configuration.html#use-in-pymc-marketing

**Comment by cluhmann** [comment]
I guess is some additional info. But where does one figure out that keys such as `"alpha"` and `"beta"` are valid? Or keys like `"intercept"`, `"saturation_beta"`, `"saturation_lam"`, and `"likelihood"`? And/or when are such key valid and when are they invalid? Are they model-dependent? Are they tied to choices about the adstock and/or saturation functions?

**Comment by cluhmann** [comment]
`beta_channel` is still used in the example [here](https://www.pymc-marketing.io/en/stable/api/generated/pymc_marketing.mmm.mmm.MMM.html), but according to [this comment](https://github.com/pymc-labs/pymc-marketing/discussions/1190#discussioncomment-11230239) I guess that's incorrect?  But this is all just further evidence (to me), that there needs to be some definitive documentation about what can go into a model config.  The scattered examples (some of which are currently incorrect/outdated) aren't cutting it.

**Comment by wd60622** [comment]
Yes, unfortunately, the model_config is just a dictionary which can have arbitrary keys. In some cases with the current structuring of them, there are dynamic keys based on other aspects of the configuration as you noted above. Most of the time the keys are defined in the `default_model_config` but in case of `BetaGeoBetaBinom` model, they are dependent on the parameterization: 

https://github.com/pymc-labs/pymc-marketing/blob/0d4e81d3eb7117e876ee84071c5272367dcb7669/pymc_marketing/clv/models/beta_geo_beta_binom.py#L174-L237
